model/riscv_hext_regs.sail

```diff
@@ -5,3 +5,293 @@
 /*                                                                                       */
 /*  SPDX-License-Identifier: BSD-2-Clause                                                */
 /*=======================================================================================*/
+
+/* Current Virtualization Mode */
+
+register cur_virtualization : Virtualization
+
+/* Hypervisor Status Register */
+
+bitfield Hstatus : xlenbits = {
+  /* VSXL only exists if HSXLEN == 64 so it's modelled via
+   * explicit an getter and setter below. */
+  /* VSXL : 33 .. 32 */
+  VTSR  : 22,
+  VTW   : 21,
+  VTVM  : 20,
+  VGEIN : 17 .. 12,
+  HU    : 9,
+  SPVP  : 8,
+  SPV   : 7,
+  GVA   : 6,
+  VSBE  : 5
+}
+register hstatus : Hstatus
+
+function get_hstatus_VSXL(h : Hstatus) -> arch_xlen = {
+  if sizeof(xlen) == 32
+  then arch_to_bits(RV32)
```

Comment:
"When HSXLEN=32, VSXL field does not exist."  The spec isn't terribly clear on what that means, but I'd assume WPRI, so the read should be 0b00 and not 0b01.

I see this code was inspired by the UXL field of the sstatus register, so I'll send a patch for that, unless you know something I missed.

Thanks.